components/com_kunena/site/controller/widget/announcement/display.php

```diff
@@ -45,8 +45,9 @@ protected function before()
 			return false;
 		}
 
-		$view = JRequest::getWord ( 'view', 'default' );
-		$layout = JRequest::getWord ( 'layout', 'default' );
+		$jinput = JFactory::getApplication()->input;
```

Comment:
You need to use $this->input, otherwise HMVC nesting breaks. Basically you can enter your own input to the controllers allowing them to render whatever you want to.